lib/SILGen/SILGenPoly.cpp

```diff
@@ -3588,13 +3741,41 @@ SILGenModule::getOrCreateAutoDiffAssociatedFunctionReorderingThunk(
       origAssocFnType->getResults().back().getSILStorageType())
           .castTo<SILFunctionType>();
 
-  auto *linearMapThunk = thunkSGF.getOrCreateAutoDiffLinearMapReorderingThunk(
-      assocFnKind, linearMapFnType, targetLinearMapFnType);
+  // Free arguments.
+  for (auto arg : argumentsToFree) {
+    if (arg->getType().isObject())
+      thunkSGF.B.emitReleaseValueAndFold(loc, arg);
+    else
+      thunkSGF.B.emitDestroyAddrAndFold(loc, arg);
```

Comment:
Here `release_value` and `destroy_addr` are the correct instructions to use, but the "and fold" part is never exercised in thunks like this. I'd suggest using builder functions `createReleaseValue` and `createDestroyAddr` instead.

Reply:
Done in https://github.com/apple/swift/pull/26380/commits/9d4bf257c138125575b97b2e2585f96376d55d85, thanks!